### PR TITLE
Feature/pelagos 5193 fix newline problem in metadata

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: templates/MetadataGenerator/MD_Distribution.xml.twig [hunk 1]
```diff
@@ -26,7 +26,7 @@
                                         {% if distPoint.dataCenter.deliveryPoint is empty %}
                                             <gco:CharacterString/>
                                         {% else %}
-                                            <gco:CharacterString><![CDATA[{{ distPoint.dataCenter.deliveryPoint }}]]></gco:CharacterString>
+                                            <gco:CharacterString>{{ distPoint.dataCenter.deliveryPoint }}></gco:CharacterString>
```
Comment (fightingtexasaggie) [inline templates/MetadataGenerator/MD_Distribution.xml.twig:29]:
```suggestion
                                            <gco:CharacterString>{{distPoint.dataCenter.deliveryPoint }}</gco:CharacterString>
```

File: src/Twig/Extensions.php [hunk 2]
```diff
@@ -5,6 +5,8 @@
 use App\Entity\DIF;
 use App\Util\MaintenanceMode;
 use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Query\Expr\Func;
```
Comment (mickel1138) [inline src/Twig/Extensions.php:8]: Not used, remove.

File: src/Twig/Extensions.php [hunk 3]
```diff
@@ -141,9 +143,29 @@ public function getFilters()
                 'orTemplateIfNotExists',
                 [$this, 'doesTwigFileExist']
             ),
+            new \Twig\TwigFilter(
+                'asXml',
```
Comment (mickel1138) [inline src/Twig/Extensions.php:147]: I'm not super happy about this name, what is a good name for this? It just formats XML, name turn it into XML.

File: src/Twig/Extensions.php [hunk 4]
```diff
@@ -141,9 +143,29 @@ public function getFilters()
...
+                [$this, 'makeXml']
+            ),
         );
     }
 
+    public function makeXml(string $xml)
```
Comment (mickel1138) [inline src/Twig/Extensions.php:153]: Ad docblock and return (string) but then maybe for $tidy to string.

File: src/Twig/Extensions.php [hunk 5]
```diff
@@ -141,9 +143,29 @@ public function getFilters()
...
+        $tidyXml = new \tidy();
+            $tidyXml->parseString(
+                $xml,
+                array(
+                    'input-xml' => true,
+                    'output-xml' => true,
+                ),
+                'utf8'
+            );
+        $xml = $tidyXml;
```
Comment (mickel1138) [inline src/Twig/Extensions.php:164]: Cast to `(string)` if return is string. There doesn't seem to be a `toString`.

File: src/Util/Metadata.php [hunk 6]
```diff
@@ -48,23 +49,27 @@ public function getXmlRepresentation(Dataset $dataset, array $boundingBoxArray)
...
+        //             'input-xml' => true,
+        //             'output-xml' => true,
+        //         ),
+        //         'utf8'
+        //     );
+        //     $xml = $tidyXml;
+        //     // Remove extra whitespace added around CDATA tags by tidy.
+        //     $xml = preg_replace('/>[\s]+<\!\[CDATA\[/', '><![CDATA[', $xml);
+        //     $xml = preg_replace('/]]>\s+</', ']]><', $xml);
         }
```
Comment (mickel1138) [inline src/Util/Metadata.php:65]: Cleanup and remove if not needed.

File: templates/MetadataGenerator/CI_ResponsibleParty.xml.twig [hunk 7]
```diff
@@ -3,7 +3,7 @@
             <gco:CharacterString>{{Person.firstName}} {{Person.lastName}}</gco:CharacterString>
         </gmd:individualName>
         <gmd:organisationName>
-            <gco:CharacterString><![CDATA[{{Person.organization | raw}}]]></gco:CharacterString>
+            <gco:CharacterString>{{Person.organization}}</gco:CharacterString>
```
Comment (mickel1138) [inline templates/MetadataGenerator/CI_ResponsibleParty.xml.twig:6]: Could maybe write an "auto" cdata maker in twig, when linebreaks are detected.

File: src/Twig/Extensions.php (hunk 3 above)
Comment (fightingtexasaggie) [reply src/Twig/Extensions.php:147]: xmlFormatter maybe?

File: src/Twig/Extensions.php [hunk 8]
```diff
@@ -5,8 +5,10 @@
 use App\Entity\DIF;
 use App\Util\MaintenanceMode;
 use Doctrine\Common\Collections\Collection;
+use DOMDocument;
 use Symfony\Component\HttpKernel\KernelInterface;
 use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use tidy;
```
Comment (mickel1138) [inline src/Twig/Extensions.php:11]: Remove, use \ where it's used.

File: templates/MetadataGenerator/CI_ResponsibleParty.xml.twig [hunk 9]
```diff
@@ -3,7 +3,7 @@
             <gco:CharacterString>{{Person.firstName}} {{Person.lastName}}</gco:CharacterString>
         </gmd:individualName>
         <gmd:organisationName>
-            <gco:CharacterString><![CDATA[{{Person.organization | raw}}]]></gco:CharacterString>
```
Comment (mickel1138) [inline templates/MetadataGenerator/CI_ResponsibleParty.xml.twig:6]: Put back?

File: templates/MetadataGenerator/CI_ResponsibleParty.xml.twig [hunk 10]
```diff
@@ -23,9 +23,7 @@
             <gmd:address>
                 <gmd:CI_Address>
                         <gmd:deliveryPoint>
-                            <gco:CharacterString>
-                                <![CDATA[{{Person.deliveryPoint | trim | raw}}]]>
-                            </gco:CharacterString>
```
Comment (mickel1138) [inline templates/MetadataGenerator/CI_ResponsibleParty.xml.twig:28]: Put back/revert.

File: src/Twig/Extensions.php [hunk 11]
```diff
@@ -141,9 +142,35 @@ public function getFilters()
...
+                [$this, 'makeXml']
+            ),
         );
     }
 
+    /**
+     * Return a cleaned-up string representation of a possibly messy string representation of XML input.
+     *
+     * @param string Text representation of XML data, possibly ugly.
+     * @return string A tidyfied cleaned-up string representation of the XML.
```
Comment (mickel1138) [inline src/Twig/Extensions.php:156]: Needs spacing between these two, but return it not needed, unless you want to describe it.

**Review by mickel1138 — APPROVED** [review]

**Pull request merged**